pkg/core/resource_graph.go

```diff
@@ -428,6 +428,37 @@ func (rg *ResourceGraph) callCreate(targetValue reflect.Value, metadata any) err
 	return nil
 }
 
+// CallConfigure uses the resource graph to ensure the node passed in exists, then uses reflection to call the resources Configure method
+func (rg *ResourceGraph) CallConfigure(resource Resource, metadata any) error {
+
+	if rg.GetResource(resource.Id()) == nil {
+		return fmt.Errorf("resource with id %s cannot be configured since it does not exist in the ResourceGraph", resource.Id())
```

Comment:
i dont know when this would be hit because the only place its used today is when we explicitly loop over the dags nodes, but in case someone uses it on a node we dont add i raised it. Adding it may have weird side effects i figured because you cant really configure something which hasnt been created and if we want to always call the resources Create method i didnt want to add it here